web-local/src/lib/util/get-table-references/test-data.ts

```diff
@@ -0,0 +1,138 @@
+import type { Reference } from "./";
+interface ReferenceTestCase {
+  query: string;
+  references: Reference[];
+}
+
+export const singleFrom: ReferenceTestCase = {
+  query: `select * FROM tbl`,
+  references: [
+    {
+      reference: "tbl",
+      type: "from",
+      index: 9,
+      referenceIndex: 14,
+    },
+  ],
+};
+
+export const bigGapInFrom: ReferenceTestCase = {
+  query: `select * FROM 
+
+
+
+
+tbl`,
+  references: [
+    {
+      reference: "tbl",
+      type: "from",
+      index: 9,
+      referenceIndex: 19,
+    },
+  ],
+};
+
+/** subqueries don't count as a reference, exactly, but the FROM statement
+ * in a subquery does.
+ */
+export const subqueryFrom: ReferenceTestCase = {
+  query: `select * from (select * from tbl)`,
+  references: [
+    {
+      reference: "tbl",
+      type: "from",
+      index: 24,
+      referenceIndex: 29,
+    },
+  ],
+};
+
+/** unfinished queries shouldn't have any references */
+export const unfinishedFrom: ReferenceTestCase = {
+  query: `select * from `,
+  references: [],
+};
+
+/** simple join. Also tests line breaks. */
+export const simpleJoin: ReferenceTestCase = {
+  query: `select * from tbl JOIN 
+  
+  x ON tbl.id = x.id`,
+  references: [
+    {
+      reference: "tbl",
+      type: "from",
+      index: 9,
+      referenceIndex: 14,
+    },
+    {
+      reference: "x",
+      type: "join",
+      index: 18,
+      referenceIndex: 29,
+    },
+  ],
+};
+
+export const remoteSourceFrom: ReferenceTestCase = {
+  query: `FROM "s3://path/to/bucket.parquet"`,
+  references: [
+    {
+      reference: '"s3://path/to/bucket.parquet"',
+      type: "from",
+      index: 0,
+      referenceIndex: 5,
+    },
+  ],
+};
+
+/** do not capture aliases (for now) */
+export const remoteSourceFromAlias: ReferenceTestCase = {
+  query: `FROM "s3://path/to/bucket.parquet" as tbl`,
+  references: [
+    {
+      reference: '"s3://path/to/bucket.parquet"',
+      type: "from",
+      index: 0,
+      referenceIndex: 5,
+    },
+  ],
+};
+
+export const remoteSourceJoin: ReferenceTestCase = {
+  query: `   FROM tbl JOIN "s3://path/to/bucket.parquet" as tbl2 ON tbl2.id = tbl.id`,
+  references: [
+    {
+      reference: "tbl",
+      type: "from",
+      index: 3,
+      referenceIndex: 8,
+    },
+    {
+      reference: '"s3://path/to/bucket.parquet"',
+      type: "join",
+      index: 12,
+      referenceIndex: 17,
+    },
+  ],
+};
+
+export const fromInCTE: ReferenceTestCase = {
+  query: `WITH x as (select * from tbl) FROM x`,
+  references: [
+    { reference: "tbl", type: "from", index: 20, referenceIndex: 25 },
+    { reference: "x", type: "from", index: 30, referenceIndex: 35 },
```

Comment:
Found the issue after merging, with the new approach returning CTE aliases we would end up marking the model dependant on any table that might have the same name as the alias.